src/systems/a380_systems/src/pneumatic.rs

```diff
@@ -639,41 +582,30 @@ impl BleedMonitoringComputerChannel {
         self.engine_starter_valve_is_open = sensors.engine_starter_valve_is_open();
 
         self.is_engine_bleed_pushbutton_auto =
-            overhead_panel.engine_bleed_pb_is_auto(self.engine_number);
+            pneumatic_overhead_panel.engine_bleed_pb_is_auto(self.engine_number);
         self.is_engine_fire_pushbutton_released = is_engine_fire_pushbutton_released;
 
-        self.is_apu_bleed_valve_open = apu_bleed_valve.is_open();
-        self.is_apu_bleed_on = overhead_panel.apu_bleed_is_on();
-
-        self.cross_bleed_valve_selector = overhead_panel.cross_bleed_mode();
-        self.cross_bleed_valve_is_open = cross_bleed_valve.is_open();
-    }
-
-    fn operation_mode(&self) -> BleedMonitoringComputerChannelOperationMode {
-        self.operation_mode
-    }
+        self.is_any_bleed_pushbutton_off = !pneumatic_overhead_panel.engine_bleed_pb_is_auto(1)
+            || !pneumatic_overhead_panel.engine_bleed_pb_is_auto(2)
+            || !pneumatic_overhead_panel.engine_bleed_pb_is_auto(3)
+            || !pneumatic_overhead_panel.engine_bleed_pb_is_auto(4);
```

Comment:
I don't know if this is better or not:
```suggestion
        self.is_any_bleed_pushbutton_off = (1..=4)
            .map(|e| !pneumatic_overhead_panel.engine_bleed_pb_is_auto(e))
            .fold(false, |a, b| a || b);
```
P.S.:
For performance reasons it would be better to have this calculated in `CoreProcessingInputOutputModuleA::update` instead but I think that's not a big issue.

Reply:
Good idea. Although I think `self.is_any_bleed_pushbutton_off = (1..=4).any(|e| !pneumatic_overhead_panel.engine_bleed_pb_is_auto(e));` is slightly more intuitive.

And yes, I agree that it would be more performant, but I am not sure how the logic works IRL. It could be that the computation is indeed run on each individual unit.

Reply:
Yeah using `any()` is even better.